Cron/UpdateOrdersToProcessing.php

```diff
@@ -0,0 +1,295 @@
+        $this->logger                   = $logger;
+    }
+
+    public function execute()
+    {
+
+        $this->logger->info("Cronjob: Update Orders To Processing Cron started.");
+
+        $dateTimeCheck = date('Y-m-d H:i:s', strtotime('-' . static::PROCESS_ORDER_WAIT_TIME . ' minutes'));
+        $sortOrder = $this->sortOrderBuilder->setField('entity_id')->setDirection('DESC')->create();
+
+        $searchCriteria = $this->searchCriteriaBuilder
+                            ->addFilter(
+                                'updated_at',
+                                $dateTimeCheck,
+                                'lt')
+                            ->addFilter(
+                                'status',
+                                static::STATUS_PENDING,
+                                'eq'
+                            )->setSortOrders(
+                                [$sortOrder]
+                            )->create();
+
+        $orders = $this->orderRepository->getList($searchCriteria);
+
+        foreach ($orders->getItems() as $order)
+        {
+            if ($order->getPayment()->getMethod() === 'razorpay') {
+                $rzpWebhookData = $order->getRzpWebhookData();
+                if (empty($rzpWebhookData) === false) // check if webhook cron has run and populated the rzp_webhook_data column
+                {
+                    $rzpWebhookDataObj = unserialize($rzpWebhookData);
+
+                    if($rzpWebhookDataObj['webhook_verified_status'] === true)
+                    {
+                        $this->updateOrderStatus($order, $rzpWebhookDataObj);
+                    } 
+                }   
+            }
+        }
+    }
+
+    private function updateOrderStatus($order, $rzpWebhookData)
+    {
+        if ($order)
+        {
+            $this->logger->info("Cronjob: Updating to Processing for Order ID: " 
+                                . $order->getEntityId() 
+                                . " and Event :" 
+                                . $rzpWebhookData['event']);
+
+            $payment = $order->getPayment();
+            $paymentId = $rzpWebhookData['payment_id'];
+            $rzpOrderAmount = $rzpWebhookData['amount'];
+            $event = $rzpWebhookData['event'];
+
+            $payment->setLastTransId($paymentId)
+                    ->setTransactionId($paymentId)
+                    ->setIsTransactionClosed(true)
+                    ->setShouldCloseParentTransaction(true);
+
+            $payment->setParentTransactionId($payment->getTransactionId());
+
+            if ($event === 'payment.authorized')
+            {
+                $payment->addTransactionCommentsToOrder(
+                    "$paymentId",
+                    (new AuthorizeCommand())->execute(
+                        $payment,
+                        $order->getGrandTotal(),
+                        $order
+                    ),
+                    ""
+                );
+            }
+            else
```

Comment:
@nikhilsharma1100 Please add `else if` condition for event `order.paid` here.

Reply:
done